src/boot/efi/boot.c

```diff
@@ -651,7 +656,7 @@ static bool menu_run(
         _cleanup_free_ char16_t *clearline = NULL, *separator = NULL, *status = NULL;
         uint32_t timeout_efivar_saved = config->timeout_sec_efivar;
         uint32_t timeout_remain = config->timeout_sec == TIMEOUT_MENU_FORCE ? 0 : config->timeout_sec;
-        bool exit = false, run = true, firmware_setup = false;
+        bool exit = false, run = true, firmware_setup = false, poweroff = false, reboot = false;
```

Comment:
maybe instead of having multiple bools clean this up and make it an enum?

```c
enum {
        ACTION_CONTINUE,
        ACTION_RUN,
        ACTION_FIRMWARE_SETUP,
        ACTION_POWEROFF,
        ACTION_REBOOT,
} action = ACTION_CONTINUE;
```

Reply:
Updated like this. I also added on more commit to rename functions, PTAL.